### PR TITLE
Add com.nolimitconnect.NoLimitConnect

### PR DESCRIPTION
**Pull request by brettrjones-github**
### Please confirm your submission meets all the criteria

- [X] Please describe your application briefly.
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: com.nolimitconnect.NoLimitConnect.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+sdk: org.kde.Sdk
+command: nolimitconnect
+finish-args: 
+    - --device=all
+    - --share=ipc
+    - --share=network
+    - --socket=fallback-x11       
+    - --socket=wayland
+    - --socket=pulseaudio
+    - --filesystem=host
```
Comment (bbhtt) [inline com.nolimitconnect.NoLimitConnect.yml:14]: host is very broad. Why do you need it?

**Comment by bbhtt** [comment]
>  Please describe your application briefly.

Can you please fill this in?

bot, build com.nolimitconnect.NoLimitConnect

File: com.nolimitconnect.NoLimitConnect.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+
+app-id: com.nolimitconnect.NoLimitConnect
+runtime: org.kde.Platform
+runtime-version: '6.7'
+sdk: org.kde.Sdk
+command: nolimitconnect
+finish-args: 
+    - --device=all
```
Comment (hfiguiere) [inline com.nolimitconnect.NoLimitConnect.yml:8]: why device all?

File: com.nolimitconnect.NoLimitConnect.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+       commit: f4bbe2c9a8cecd72a2f6cf90e881c744ba6e618d
+       tag: v1.0.5
+
+     
+     # for local sources so does not do a git pull each time
+     # - type: dir
+     #   path: ./
+
+
+
```
Comment (hfiguiere) [inline com.nolimitconnect.NoLimitConnect.yml:41]:
```suggestion
```

**Comment by brettrjones-github** [comment]
Partially this is from
https://github.com/flathub/org.qbittorrent.qBittorrent/blob/master/org.qbittorrent.qBittorrent.yaml
but I am not absolutely certain I need all of host. What I need is the
ability to read all media types like videos, pictures music etc for user to
add to library and/or share
I do not need write access anywhere except in the normal sandbox for
downloading and media caching etc.
What do you think? can I get read access anywhere in home without the
filesystem=host

File: com.nolimitconnect.NoLimitConnect.yml (hunk 2 above)
Comment (brettrjones-github) [reply com.nolimitconnect.NoLimitConnect.yml:8]: It is multimedia ie camera, microphone recording/playback etc. There does not seem to be a specification for these.

File: com.nolimitconnect.NoLimitConnect.yml (hunk 3 above)
Comment (brettrjones-github) [reply com.nolimitconnect.NoLimitConnect.yml:41]: Will remove in my next commit

File: com.nolimitconnect.NoLimitConnect.yml (hunk 1 above)
Comment (brettrjones-github) [reply com.nolimitconnect.NoLimitConnect.yml:14]: Partially this is from https://github.com/flathub/org.qbittorrent.qBittorrent/blob/master/org.qbittorrent.qBittorrent.yaml
but I am not absolutely certain I need all of host. What I need is the ability to read all media types like videos, pictures music etc from anywhere the user has access to for user to add to library and/or share
I do not need write access anywhere except in the normal sandbox for downloading and media caching etc.
What do you think? can I get read access anywhere in home without the filesystem=host

**Comment by brettrjones-github** [comment]
Ok I see the use of <br/> caused issues. will remove
As for the description of the app it is not a simple app. Designed for a distributed and scalable social network were for a small group a cell with internet access is enough to do all hosting and host listing etc.
As for features there is alot of them. Some of the more unique ones is push to talk, web cam service, video streaming. Here is what I thought I put in the description.
    <description>
        <p>NoLimitConnect is a full featured social network.</p>
        <p>A independent social network designed to be hosted on any device with an internet connection.</p>
        <p>Major features:<br/>
        o Person to Person and plugin permissions: Ignore, Guest, Friend, Admin<br/>
	    o Push To Talk VOIP<br/>
		o Stream shared media files<br/>
		o Share and transfer files (No file size limit)<br/>
        o Video Chat<br/>
        o VOIP Voice Phone<br/>
        o Messenger<br/>
        o About Me user page<br/>
        o Storyboard blog page<br/>
        o Web Cam Server<br/>>
        o Media Player (Based on Kodi(TM) and Ffmpeg)<br/>
        o Chat Room host and client<br/>
        o Random Connect host and client<br/>
        o Group host and client</p>
    </description>

File: com.nolimitconnect.NoLimitConnect.yml (hunk 1 above)
Comment (bbhtt) [reply com.nolimitconnect.NoLimitConnect.yml:14]: Does it support filechooser portals ? https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.FileChooser.html

Comment (brettrjones-github) [reply com.nolimitconnect.NoLimitConnect.yml:14]: It does not support filechooser portals. It uses mostly c style fopen fread fclose style file io and some use of Qt QFile. It compiles on Windows/Linux/Android which means as generic of code as possible mostly.

**Comment by brettrjones-github** [comment]
I there a way to tell flatpak not to build arm builds. It only supports x86_64 for Windows and Linux and for Android it only supports ARM.

**Comment by bbhtt** [comment]
> I there a way to tell flatpak not to build arm builds. It only supports x86_64 for Windows and Linux and for Android it only supports ARM.

https://docs.flathub.org/docs/for-app-authors/maintenance#limiting-the-set-of-architectures-to-build-on

You have to fix the metainfo issues reported too.

**Comment by brettrjones-github** [comment]
version 1.0.6
1.) Removed br tag from metainfo description
2.) Bump version from 1.0.5 to 1.0.6
3.) Remove path comments from com.nolimitconnect.NoLimitConnect.yml
4.) Added only-arches: x86_64

File: com.nolimitconnect.NoLimitConnect.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+app-id: com.nolimitconnect.NoLimitConnect
+runtime: org.kde.Platform
+runtime-version: '6.7'
+sdk: org.kde.Sdk
+only-arches: x86_64
```
Comment (bbhtt) [inline com.nolimitconnect.NoLimitConnect.yml:6]:
```suggestion
```

This is not supposed to be here. Please see the docs again, it should be in a flathub.json file.

**Comment by bbhtt** [comment]
These bullet points should use `li` https://gitlab.com/nolimitconnectapps/nolimitapp/-/commit/5031fdeeeef180a4998da35216e6e78b182f1c66#a78d355910c40debee54e8e2680fad43809f89a5_25_12 

https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#description

You can preview the metainfo file before submitting https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#preview

File: com.nolimitconnect.NoLimitConnect.yml (hunk 4 above)
Comment (brettrjones-github) [reply com.nolimitconnect.NoLimitConnect.yml:6]: So does that mean I cannot use yml and have to use json or is this an additional file

**Comment by brettrjones-github** [comment]
gnome-software --show-metainfo ./com.nolimitconnect.NoLimitConnect.metainfo.xml

Unknown option --show-metainfo

**Comment by brettrjones-github** [comment]
version 1.0.7
1.) use li requirement in metainfo file
2.) Bump version from 1.0.6 to 1.0.7
3.) Remove only-arches: x86_64 from yml file
4.) Add flathub.json with "only-arches": ["x86_64"]

**Comment by bbhtt** [comment]
You are on an old version probably. What version of GNOME Software do you have? 

image

**Comment by bbhtt** [comment]
bot, build com.nolimitconnect.NoLimitConnect

**Comment by hfiguiere** [comment]
> I there a way to tell flatpak not to build arm builds. It only supports x86_64 for Windows and Linux and for Android it only supports ARM.

Why?

**Comment by bbhtt** [comment]
https://gitlab.com/nolimitconnectapps/nolimitapp/-/blob/8a60b320888cf5675a6b44fd40e713bbabd713af/com.nolimitconnect.NoLimitConnect.metainfo.xml#L43 does not match the app id here.

**Comment by brettrjones-github** [comment]
gnome-software --version

gnome-software 41.5

**Comment by brettrjones-github** [comment]
version 1.0.8
1.) fix app id in metainfo file
2.) Bump version from 1.0.7 to 1.0.8

**Comment by brettrjones-github** [comment]
Why not do all cpu architectures? Lack of time and hardware. I have no Linux arm device to test with and even if I did I would probably limit my scope to just the most used. I still need to get to the bug fix/feature bloat/Scale testing phase

**Comment by hfiguiere** [comment]
it's clear that no else will test it in that situation. Even there don't seem to be any reason to cut the architecture off like that.

**Comment by brettrjones-github** [comment]
Can I get a status update? Is there something I missed and still need to do? It is not a requirement that it builds for both x64 and arm is it?

**Comment by bbhtt** [comment]
If you can, I suggest building it for aarch64.

**Comment by brettrjones-github** [comment]
I ordered a rasberry pi 5 kit. I will build it for rasberry pi and then hopefully it will also build for flathub. It will be at minimum 3 weeks. I will post a update to my com.nolimitconnect.NolimitConnect branch when I think it will build

**Comment by brettrjones-github** [comment]
I am trying to build on raspberry pi. I added to flathub.json
{
    "only-arches": ["aarch64", "x86_64"]
}
I need a way to detect within CMake if it is being built for x64 or aarch64 so that the correct compile defines can be set. There does not seem to be any params passed to cmake except the normal 
CMakeArgs:  -DCMAKE_BUILD_TYPE="Release" -DCMAKE_INSTALL_PREFIX:PATH="/app" 

QCreator passes -DANDROID_ABI:STRING="arm64-v8a" for android build. Is there some kind of equivalent I can use for flathub builds?

**Comment by hfiguiere** [comment]
consult CMake documentation for that question. It knows how do detect the architecture. This is NOT cross compilation either, so in general this is not needed unless there are bits that (wrongly) assume a particular architecture.

**Comment by brettrjones-github** [comment]
version 1.0.9
compiles for both x86 and aarch64 CPU

File: flathub.json
Comment (bbhtt) [inline flathub.json:1]: This file is not needed any more.

Comment (brettrjones-github) [reply flathub.json:1]: removed flathub.json

**Comment by bbhtt** [comment]
bot, build com.nolimitconnect.NoLimitConnect

**Comment by bbhtt** [comment]
Is it working? Seems like the app closes/crashes once the login dialogue appears and I click on some buttons.

**Comment by brettrjones-github** [comment]
Well It works on windows, ubuntu 22.04 and raspberry pi 5 and android. So there must be a sequence of events or something. Did you try just enter name and press login?

**Comment by brettrjones-github** [comment]
Most likely it is some sandbox permission error. That is what is different with flatpak. Is there a way I can simulate the same environment as you are?

**Comment by brettrjones-github** [comment]
I ran using flatpak run com.nolimitconnect.NoLimitConnect on ubuntu 22.04 and it worked fine for me. I have no way to debug this without more info

**Comment by bbhtt** [comment]
Yea I can't reproduce the crash anymore. But how does the application work? I entered a login name and connected, but join group/join chat seems empty?

image

**Comment by bbhtt** [comment]
If you can screenrecord a 5-10s clip to show how it works, that'd be appreciated. The interface is a bit confusing to me.

**Comment by brettrjones-github** [comment]
First thanks for insisting on the arm build. That will make it easier for setting up low cost hosts. Second there is no users except me and I am mostly busy with my job and this hobby project. I have friends that will help test and enhance the application but they are not technical people so I need to publish and then they can simply download and run. 

There are many good large scale social networks like facebook and linkedin etc. if you do not mind the constant spam and requests to connect and public scrutiny like facebook jail etc. That is not the intent of this application. The intent is one or more network hosts. The network host is just a listing service for active user hosts and how to connect to them (assuming you have sufficient permission to join that particular user host). There is no separate applications for servers this is all in one, even the NoLimitConnect (NLC) network host is just this application running on a ubuntu VPS but with network host enabled. 

For example I was planning to setup a Group/Chatroom/Random Connect hosts for testing on an android tablet. Now that I have a raspberry pi I plan to enable those hosts on it and enable a web cam and storyboard and serve some videos on the raspberry pi for permanent user host testing.

For your particular can I can see by the signal bars you do not have a open port so you cannot setup any hosting. I recommend hide.me VPN to open whatever port you network settings are set to then enable chat room or whatever hosting you want then with another device you can connect to your host and talk to yourself or have others join you. Devices that are not hosts do not require a open port. 

So this is kind of a chicken and egg thing. I need to publish so I can have testers help me test and so that others can join.

Also now that you mention it I need to update the website to explain all this. It is very different from a traditional large scale social network.

**Comment by brettrjones-github** [comment]
updated website https://nolimitconnect.com with a What is No Limit Connect section on the home page

**Comment by brettrjones-github** [comment]
The website source code is also open source
git clone https://github.com/brettrjones-github/nlc-nolimitconnect.com.git
I use Visual Studio Code to compile and commit the website.
I am not a web programmer. I mostly learned how to build this website from this video.
https://www.youtube.com/watch?v=4wD00RT6d-g

**Comment by bbhtt** [comment]
> So this is kind of a chicken and egg thing. I need to publish so I can have testers help me test and so that others can join.

I guess this can be added as a beta app to Flathub. You can move it to stable once it is ready.

**Comment by brettrjones-github** [comment]
That works for me. BTW how do I give NLC read permission to the whole home directory. I tried to add a video to stream from my Downloads folder but it is greyed out. When I run it from QCreator it works fine

**Comment by bbhtt** [comment]
It should already have that. you have whole host access which includes full home directory too

**Comment by brettrjones-github** [comment]
Beta is fine with me.

**Comment by brettrjones-github** [comment]
It looks like I already have some changes I need to make. Do I go through this process again or how do I update it?

**Comment by bbhtt** [comment]
update the commit here by pushing?

**Comment by brettrjones-github** [comment]
I found a bug I need to fix. so I will fix it and bump the version. I found how to use the beta repo but not how to mark my submission as beta. Could you post a link of how to change this submission to a beta submission?

**Comment by bbhtt** [comment]
You don't need to do anything, we'll merge it for `beta` branch. You can just bump the version.

**Comment by brettrjones-github** [comment]
Thanks. I will fix that bug and bump the version then push

**Comment by brettrjones-github** [comment]
I think Maybe something happened to new-pr branch since I took so long
git clone --branch=new-pr git@github.com:brettrjones-github/flathub.git
cd flathub
ls
shows no files
I am ready to publish to beta

version 1.0.10
1.) Improve media player
2.) Major rework of project to remove as many dependency libraries and configuration files as possible

Is my checkout steps wrong or is there something else I need to do

**Comment by brettrjones-github** [comment]
revision is 5e52159e573bcf3695721560145556829d468d09

**Comment by brettrjones-github** [comment]
Tag is v1.0.10

**Comment by brettrjones-github** [comment]
I figured out what I did wrong and updated and pushed the new tag/revision
Should be a shorter compile time now also

**Comment by bbhtt** [comment]
You haven't pushed any changes here. Make sure to checkout your branch, not `new-pr`

**Comment by brettrjones-github** [comment]
it is pushed now. double checked by doing fresh checkout in new folder

**Comment by bbhtt** [comment]
bot, build com.nolimitconnect.NoLimitConnect

**Comment by hfiguiere** [comment]
> > Please describe your application briefly.
> 
> Can you please fill this in?


This is still not done.

File: com.nolimitconnect.NoLimitConnect.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    - --share=network
+    - --socket=fallback-x11       
+    - --socket=wayland
+    - --socket=pulseaudio
+    - --filesystem=host
+
+modules:
+    ########################
+    # GLU 
+    ########################
```
Comment (hfiguiere) [inline com.nolimitconnect.NoLimitConnect.yml:19]:
```suggestion
```

File: com.nolimitconnect.NoLimitConnect.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+ - shared-modules/glu/glu-9.json
+    ########################
+    # No Limit Connect
+    ########################
```
Comment (hfiguiere) [inline com.nolimitconnect.NoLimitConnect.yml:23]:
```suggestion
```

File: com.nolimitconnect.NoLimitConnect.yml [hunk 7]
```diff
@@ -0,0 +1,36 @@
+
```
Comment (hfiguiere) [inline com.nolimitconnect.NoLimitConnect.yml:1]:
```suggestion
```

**Comment by brettrjones-github** [comment]
I pushed the suggested yml format changes
As to the description. I am assuming you want a metainfo file. Some repos have it and some do not.  I would rather avoid it if possible. Here is the reasons why
1.) How do you describe an app that has blog, web cam server, video streaming, video chat, truth or dare game, file shredder etc.
The target audience is probably people who do not like companies and spies using their personal information without permission. I would call it a DIY social network but even that is not 100% true. Although it has zero dependencies on 3rd party services, I have set up a network host on nolimitconnect.net because almost no one is going to buy a static ip and set up a network host. The network host is just this program configure as network host. A network host provides connection test services and listing of available chat rooms, group hosts, random connect hosts (These hosts are hosted by users).

2.) All code is in the same repo and has many different licenses (all licenses are open source licenses).
Here is hopefully a complete list as listed in LEGAL
= LEGAL NOTICE INFORMATION
--------------------------

All the files in this distribution are covered under either the Brett R. Jones's
license (see the file COPYING) or public-domain or the various open source projects license

No Limit Connect Author Brett R. Jones has dual license for files copyrighted by
Brett R. Jones and is located at nolimitapp/COPYING

Here is a listing of LICENSE locations for your convenience.
This might not be a complete list.

 nolimitapp/LICENSE
 nolimitapp/nolimitgui/CommonSrc/Resources/CreditsArt.txt

 
 nolimitapp/nolimitlib/DependLibs/bzlib/LICENSE
 nolimitapp/nolimitlib/DependLibs/ffmpeg/LICENSE.md
 nolimitapp/nolimitlib/DependLibs/ffmpeg/opus/LICENSE_PLEASE_READ.txt
 nolimitapp/nolimitlib/DependLibs/googletest/googlemock/LICENSE
 nolimitapp/nolimitlib/DependLibs/googletest/googlemock/scripts/generator/LICENSE
 nolimitapp/nolimitlib/DependLibs/googletest/googletest/LICENSE
 nolimitapp/nolimitlib/DependLibs/libaom/LICENSE
 nolimitapp/nolimitlib/DependLibs/libaom/third_party/fastfeat/LICENSE
 nolimitapp/nolimitlib/DependLibs/libaom/third_party/libwebm/LICENSE.TXT
 nolimitapp/nolimitlib/DependLibs/libaom/third_party/x86inc/LICENSE
 nolimitapp/nolimitlib/DependLibs/libarmrwbenc/MODULE_LICENSE_APACHE2
 nolimitapp/nolimitlib/DependLibs/libarmrwbenc/common/MODULE_LICENSE_APACHE2
 nolimitapp/nolimitlib/DependLibs/libarmrwbenc/SampleCode/MODULE_LICENSE_APACHE2
 nolimitapp/nolimitlib/DependLibs/libbzip2/LICENSE
 nolimitapp/nolimitlib/DependLibs/libcrossguid/LICENSE
 nolimitapp/nolimitlib/DependLibs/libcurl/docs/LICENSE-MIXING
 nolimitapp/nolimitlib/DependLibs/libdsp-webrtc/WEBRTC_LICENSE.txt
 nolimitapp/nolimitlib/DependLibs/libdsp-webrtc/WEBRTC_LICENSE_THIRD_PARTY.txt
 nolimitapp/nolimitlib/DependLibs/libdsp-webrtc/abseil/LICENSE
 nolimitapp/nolimitlib/DependLibs/libdsp-webrtc/common_audio/third_party/ooura/LICENSE
 nolimitapp/nolimitlib/DependLibs/libdsp-webrtc/common_audio/third_party/spl_sqrt_floor/LICENSE
 nolimitapp/nolimitlib/DependLibs/libdsp-webrtc/modules/third_party/fft/LICENSE
 nolimitapp/nolimitlib/DependLibs/libdsp-webrtc/modules/third_party/g711/LICENSE
 nolimitapp/nolimitlib/DependLibs/libdsp-webrtc/modules/third_party/g722/LICENSE
 nolimitapp/nolimitlib/DependLibs/libdsp-webrtc/modules/third_party/portaudio/LICENSE
 nolimitapp/nolimitlib/DependLibs/libdsp-webrtc/rtc_base/third_party/base64/LICENSE
 nolimitapp/nolimitlib/DependLibs/libdsp-webrtc/rtc_base/third_party/sigslot/LICENSE
 nolimitapp/nolimitlib/DependLibs/libechocancel/WEBRTC_LICENSE.txt
 nolimitapp/nolimitlib/DependLibs/libechocancel/WEBRTC_LICENSE_THIRD_PARTY.txt
 nolimitapp/nolimitlib/DependLibs/libfdk-aac/MODULE_LICENSE_FRAUNHOFER
 nolimitapp/nolimitlib/DependLibs/libfmt/LICENSE.rst
 nolimitapp/nolimitlib/DependLibs/libfreetype/docs/LICENSE.TXT
 nolimitapp/nolimitlib/DependLibs/libglew/LICENSE.txt
 nolimitapp/nolimitlib/DependLibs/libgnu/LICENSES
 nolimitapp/nolimitlib/DependLibs/libgnu/accelerated/x86/license.txt
 nolimitapp/nolimitlib/DependLibs/libgnu/extras/licenses
 nolimitapp/nolimitlib/DependLibs/libgnu/extras/licenses/CC0
 nolimitapp/nolimitlib/DependLibs/liblame/LICENSE
 nolimitapp/nolimitlib/DependLibs/liblcms2/utils/jpgicc/LICENSE_iccjpeg
 nolimitapp/nolimitlib/DependLibs/libmdnsresponder/LICENSE
 nolimitapp/nolimitlib/DependLibs/libminiaudio/LICENSE
 nolimitapp/nolimitlib/DependLibs/libminiupnpc/LICENSE
 nolimitapp/nolimitlib/DependLibs/libopencore-amr/LICENSE
 nolimitapp/nolimitlib/DependLibs/libopenh264/LICENSE
 nolimitapp/nolimitlib/DependLibs/libopenjpeg/LICENSE
 nolimitapp/nolimitlib/DependLibs/libopenjpeg/thirdparty/astyle/LICENSE.md
 nolimitapp/nolimitlib/DependLibs/libopenjpeg/thirdparty/libpng/LICENSE
 nolimitapp/nolimitlib/DependLibs/libopenmpt/LICENSE
 nolimitapp/nolimitlib/DependLibs/libopenmpt/include/foobar2000sdk/sdk-license.txt
 nolimitapp/nolimitlib/DependLibs/libopenmpt/include/foobar2000sdk/pfc/pfc-license.txt
 nolimitapp/nolimitlib/DependLibs/libopenmpt/include/htmlhelp/license.txt
 nolimitapp/nolimitlib/DependLibs/libopenmpt/include/lame/LICENSE
 nolimitapp/nolimitlib/DependLibs/libopenmpt/include/portaudio/LICENSE.txt
 nolimitapp/nolimitlib/DependLibs/libplayer-nlc/LICENSE.GPL
 nolimitapp/nolimitlib/DependLibs/libplayer-nlc/xbmc/LICENSE.GPL
 nolimitapp/nolimitlib/DependLibs/libplayer-nlc/xbmc/fmt/LICENSE.rst
 nolimitapp/nolimitlib/DependLibs/libplayer-nlc/xbmc/interfaces/json-rpc/schema/license.txt
 nolimitapp/nolimitlib/DependLibs/librapidjson/license.txt
 nolimitapp/nolimitlib/DependLibs/librapidjson/contrib/natvis/LICENSE
 nolimitapp/nolimitlib/DependLibs/libtheora/LICENSE
 nolimitapp/nolimitlib/DependLibs/libUPnP/Neptune/LICENSE
 nolimitapp/nolimitlib/DependLibs/libUPnP/Platinum/LICENSE.txt
 nolimitapp/nolimitlib/DependLibs/libvpx/LICENSE
 nolimitapp/nolimitlib/DependLibs/libvpx/third_party/libwebm/LICENSE.TXT
 nolimitapp/nolimitlib/DependLibs/libvpx/third_party/libyuv/LICENSE
 nolimitapp/nolimitlib/DependLibs/libvpx/third_party/x86inc/LICENSE
 nolimitapp/nolimitlib/DependLibs/libvpx-win64/LICENSE
 nolimitapp/nolimitlib/DependLibs/libvpx-win64/third_party/googletest/src/LICENSE
 nolimitapp/nolimitlib/DependLibs/libvpx-win64/third_party/libwebm/LICENSE.TXT
 nolimitapp/nolimitlib/DependLibs/libvpx-win64/third_party/libyuv/LICENSE
 nolimitapp/nolimitlib/DependLibs/libvpx-win64/third_party/x86inc/LICENSE
 nolimitapp/nolimitlib/DependLibs/libvpx_win/LICENSE
 nolimitapp/nolimitlib/DependLibs/libvpx_win/third_party/libwebm/LICENSE.TXT
 nolimitapp/nolimitlib/DependLibs/libvpx_win/third_party/x86inc/LICENSE
 nolimitapp/nolimitlib/DependLibs/libwavepack/license.txt
 nolimitapp/nolimitlib/DependLibs/libx265/dynamicHDR10/LICENSE.txt
 nolimitapp/nolimitlib/DependLibs/libx265/dynamicHDR10/json11/LICENSE.txt
 nolimitapp/nolimitlib/DependLibs/libxvid/LICENSE
 nolimitapp/nolimitlib/DependLibs/libxvid/vfw/LICENSE
 nolimitapp/nolimitlib/DependLibs/libzlib/contrib/dotzlib/LICENSE_1_0.txt
 
 Tell me what I have to do.

**Comment by bbhtt** [comment]
> As to the description. I am assuming you want a metainfo file. Some repos have it and some do not. I would rather avoid it if possible. Here is the reasons why

A metainfo file is mandatory, you can describe it however you want as long as its reasonable and related to the app its fine. ALl the required metadata is described here https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#header

**Comment by brettrjones-github** [comment]
So what about the comment
[hfiguiere](https://github.com/hfiguiere) commented [Aug 18, 2024](https://github.com/flathub/flathub/pull/5254#issuecomment-2295248096)

        Please describe your application briefly.

    Can you please fill this in?

This is still not done.

Is there still something missing? 
I would like to complete the flathub beta then update android and publish the playstore beta then the real testing can begin

**Comment by bbhtt** [comment]
OK that was about the PR template in your opening post.

**Comment by bbhtt** [comment]
bot, build com.nolimitconnect.NoLimitConnect

**Comment by brettrjones-github** [comment]
Thank you for your patience. I will probably have many updates as I bug fix during beta phase. Any advice?

**Pull request closed without merging**